packages/ec-core/src/operator/selector/lexicase.rs

```diff
@@ -95,3 +100,181 @@ where
             .context("The pool of candidates was empty")
     }
 }
+
+#[cfg(test)]
+#[rustversion::attr(before(1.81), allow(clippy::unwrap_used))]
+#[rustversion::attr(
+    since(1.81),
+    expect(
+        clippy::unwrap_used,
+        reason = "Panicking is the best way to deal with errors in unit tests"
+    )
+)]
+mod tests {
+    use prop::collection;
+    use proptest::prelude::*;
+
+    use super::*;
+    use crate::individual::ec::EcIndividual;
+
+    #[test]
+    fn single_best_single_error() {
+        // 9 is the sole winner
+        let population = vec![5, 8, 9, 6, 3, 2, 0]
+            .into_iter()
+            .enumerate()
+            .map(|(id, e)| EcIndividual::<usize, TestResults<i32>>::new(id, TestResults::from([e])))
+            .collect::<Vec<_>>();
+
+        let lexicase = Lexicase::new(1);
+        let mut rng = rand::thread_rng();
+
+        assert_eq!(&2, lexicase.select(&population, &mut rng).unwrap().genome());
+    }
+
+    #[test]
+    fn multiple_copies_best() {
+        // The two 9s are the possible winners
+        let population = vec![5, 8, 9, 6, 3, 2, 0, 9]
+            .into_iter()
+            .enumerate()
+            .map(|(id, e)| EcIndividual::<usize, TestResults<i32>>::new(id, TestResults::from([e])))
+            .collect::<Vec<_>>();
+
+        let lexicase = Lexicase::new(1);
+        let mut rng = rand::thread_rng();
+
+        let selected = *lexicase.select(&population, &mut rng).unwrap().genome();
+        assert!(
+            selected == 2usize || selected == 7usize,
+            "genome {selected} should have been 2 or 7"
+        );
+    }
+
+    #[test]
+    fn single_best_multiple_errors() {
+        // [9, 8] is the sole winner
+        let population = vec![[5, 3], [8, 2], [9, 8], [6, 2], [3, 8], [2, 8], [0, 6]]
+            .into_iter()
+            .enumerate()
+            .map(|(id, es)| EcIndividual::<usize, TestResults<i32>>::new(id, TestResults::from(es)))
```

Comment:
I would probably do this instead, might be more readable:
```suggestion
            .map(TestResults::<i32>::from)
            .enumerate()
            .map(EcIndividual::new)
```

Reply:
Adding the `From<(G, R)>` implementation was pretty straightforward. What I ended up with, though, was:

```rust
        let population = [5, 8, 9, 6, 3, 2, 0]
            .into_iter()
            .map(once)
            .map(TestResults::<i32>::from)
            .enumerate()
            .map(EcIndividual::from)
            .collect::<Vec<_>>();
```

and I'm not entirely sure which approach I prefer now. This avoids any closures, which is nice, but it's not super obvious why `.map(once)` (or `.map(|v| [v])` if you prefer) is there. Maybe I should write some helper functions that encapsulate (and document) this logic rather than repeating it across multiple tests?